### ISSUE TITLE
Payment Method "PowerPay"

### ISSUE THREAD

**Issue by 3d-printerstore** [pfpayments/jtl-5#25]
Hi there
An other issue - we would have "PowerPay" (payment with invoice) as a payament option for our customers, but it does not seem to be forwarded to the payment page where the customer should choose his way of payment.

Looking forward to a solution as well.

Br's
Ralph

**Comment by Alcpaone** [comment]
@3d-printerstore how did you test?

If we try, it works fine. Also the new button is there (stornieren).

<img width="745" alt="image" src="https://github.com/pfpayments/jtl-5/assets/64169279/ee809b72-8156-461c-8a8b-e38a7e16a221">

**Comment by 3d-printerstore** [comment]
Hi

Yes, you are right – it was my fault, as I didn’t checked the payment method box in the shipping methods.

But – I find it kind of strange and out of date,  to ask for a gender “Male” or “female” and just the birth of date.



Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: Alcpaone ***@***.***>
Gesendet: Mittwoch, 10. Januar 2024 11:03
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


@3d-printerstore<https://github.com/3d-printerstore> how did you test?

If we try, it works fine. Also the new button is there (stornieren).
image.png (view on web)<https://github.com/pfpayments/jtl-5/assets/64169279/ee809b72-8156-461c-8a8b-e38a7e16a221>

**Comment by Alcpaone** [comment]
Yes a bit Strange because the Informationen are all there (our customers have to Register).

I would suggest to the developer, to autofill the Information if they are known and bypass this form. Anyway this implementation is less usefull then before.

the dedicated powerpay plugin always did the check in the background so that the customer only saw this payment option if available for him in the checkout of the shop. Now all customers can see this payment option, also customer that aren’t legit to use it.

this is the reason why a button to change payment method is so important and also a clear “error” message, that the customer was declined.

**Comment by 3d-printerstore** [comment]
I wonder how the customer would get an invoice – I don’t want to test it and then receive an invoice from PowerPay 😊


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: Alcpaone ***@***.***>
Gesendet: Mittwoch, 10. Januar 2024 13:53
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


Yes a bit Strange because the Informationen are all there (our customers have to Register).

I would suggest to the developer, to autofill the Information if they are known and bypass this form. Anyway this implementation is less usefull then before.

the dedicated powerpay plugin always did the check in the background so that the customer only saw this payment option if available for him in the checkout of the shop. Now all customers can see this payment option, also customer that aren’t legit to use it.

this is the reason why a button to change payment method is so important and also a clear “error” message, that the customer was declined.

**Comment by Alcpaone** [comment]
@3d-printerstore you can't just use it. Your company needs a contract with powerpay. After that your option is ready in the postfinance checkout and you can activate the payment option. Powerpay sends the invoice over email. All Details and how to's are in the paperwork you have to sign ;-) Works fine.

**Comment by 3d-printerstore** [comment]
ok – well, we have the contract alright, but it just looked so “off” when a customer needs to put in Gender & BirthDate


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: Alcpaone ***@***.***>
Gesendet: Mittwoch, 10. Januar 2024 14:38
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


@3d-printerstore<https://github.com/3d-printerstore> you can't just use it. Your company needs a contract with powerpay. After that your option is ready in the postfinance checkout and you can activate the payment option. Powerpay sends the invoice over email. All Details and how to's are in the paperwork you have to sign ;-) Works fine.

**Comment by Alcpaone** [comment]
ok, then you should know how your customers get the invoice, you signed the contract ;-)
anyway, yes it is not neccessary if the informations are there. If they aren't, they nedd this for verfification.
Do you ask your customers about birthday and gender?

**Comment by 3d-printerstore** [comment]
No, we would never ask our customers for their birthday or gender  - birthday would be an option for congratulating and giving out vouchers, but gender is really outdated.


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: Alcpaone ***@***.***>
Gesendet: Mittwoch, 10. Januar 2024 15:08
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


ok, then you should know how your customers get the invoice, you signed the contract ;-)
anyway, yes it is not neccessary if the informations are there. If they aren't, they nedd this for verfification.
Do you ask your customers about birthday and gender?

**Comment by Alcpaone** [comment]
@3d-printerstore "but gender is really outdated." maybe, but this is something you have to deal if you like to offer invoice without taking the risk of payment loss with powerpay. Let me know if you find a swiss company, that don't ask for that.

**Comment by 3d-printerstore** [comment]
I know , but dont‘ think I have the time looking for an another one 😉


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: Alcpaone ***@***.***>
Gesendet: Donnerstag, 11. Januar 2024 09:35
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


@3d-printerstore<https://github.com/3d-printerstore> "but gender is really outdated." maybe, but this is something you have to deal if you like to offer invoice without taking the risk of payment loss with powerpay. Let me know if you find a swiss company, that don't ask for that.

**Comment by Alcpaone** [comment]
just for information if someone struggles about the settings for powerpay.

Set this option like below in the postfinance checkout / Einstellungen / Zahlung / Zahlungsoptionen / Verarbeiter

Thats the same as the "old" plugins has done before. Customers see only payment option with invoice if credit check is ok.

<img width="722" alt="image" src="https://github.com/pfpayments/jtl-5/assets/64169279/01b43fda-ec88-41de-a00a-9957de2cd14f">

**Comment by Alcpaone** [comment]
update: the function for pre-checking creditworthiness does not work in plugin version 1.0.20. As usual, it is not known when this important function will be fixed.

**Comment by 3d-printerstore** [comment]
PowerPay told me to check the box "Nach der Wahl der Zahlungsmethode".
The problem you'll have then in the JTL Wawi , in the case of a declined customer, is an order which is not payed, but with "invoice" for a customer who would only get "prepayment"

Had to make a workaround with a Workflow which double checks those and change them to "Prepayment* - or does anybody know how to get those PowerPay orders come into the JTL Wawi as PowerPay and not as "invoice"?

**Comment by thibault-mambour** [comment]
hi @Alcpaone @3d-printerstore , could you provide us how to reproduce the problem from the setup and what is happening vs what you expect to happen ?

**Comment by Alcpaone** [comment]
@thibault-mambour 

Hi Thibault

that depends on what we are talking about here.

1. credit check BEFORE payment method selection, this problem is open as a separate issue, see #31 -> very important. The available options in the checkout obviously do not work at the moment.
![image](https://github.com/pfpayments/jtl-5/assets/64169279/2c2e2c17-074f-465a-ad8e-442957e8edec)


2. additional query of the date of birth and gender:
If this information is passed from the Online Shop to Checkout (gender, date of birth), no additional query should appear. 
If the data is not transferred because not all stores request this information, the additional query should appear as is currently the case.

**Comment by thibault-mambour** [comment]
hi @Alcpaone , do you think you could provide some screen recording, so it would be easier for us to understand ?

**Comment by Alcpaone** [comment]
@thibault-mambour I’m not Shure what you like to see in the recording. Before a customer can pay (with Powerpay) at the moment he has to fill his data again in the screen from your plugin. See the screenshot:

![image](https://github.com/pfpayments/jtl-5/assets/64169279/549fd63f-9fcd-4cc1-8094-4df9040eaa2b)

All information requested are already provided from the customer to the shop. If the information are available, the form should not appear. If the information are not provided, then the form hast to show up.

please explain exactly what you like to see, i will do that for you asap.

ps: sorry, looks like I missed your message a few days ago. Next time if you don’t get answer in one day send me please a email. You already know my adress.

**Comment by thibault-mambour** [comment]
Hi @Alcpaone , in https://github.com/pfpayments/jtl-5/issues/25#issuecomment-1959498496, you mention 2 points. For the point 1, could you provide a bit more context on what you are expecting from the plugin ? Or some explanation on how you are using it

**Comment by Alcpaone** [comment]
@thibault-mambour inserted answer in #31 where it belongs.

**Comment by 3d-printerstore** [comment]
Hi there

Can you please look at this again – I can’t recall which issue number it was, but this error was once already solved, but with the newest update it has come back
Discounts are not deducted, but added to the payment amount !!

***@***.***


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

**Comment by thibault-mambour** [comment]
hi @3d-printerstore can you check with 1.0.23, it should be fixed there

**Comment by 3d-printerstore** [comment]
@ thibault-mambour
Looks like the discount is working again - thanks


Freundliche Grüsse / Salutations / Best regards

Ralph Borbeck
3D-PRINTERSTORE.CH

except ag
Weststrasse 15
CH - 8570 Weinfelden

+41 71 620 32 38
***@***.******@***.***>
https://3d-printerstore.ch<https://3d-printerstore.ch/>

Von: thibault-mambour ***@***.***>
Gesendet: Dienstag, 5. März 2024 14:44
An: pfpayments/jtl-5 ***@***.***>
Cc: 3D Printerstore ***@***.***>; Mention ***@***.***>
Betreff: Re: [pfpayments/jtl-5] Payment Method "PowerPay" (Issue #25)


hi @3d-printerstore<https://github.com/3d-printerstore> can you check with 1.0.23, it should be fixed there

**Comment by Alcpaone** [comment]
@PAO1908CH done? if so, please close the issue so we can all see the remaining tasks. Thank you!

**Comment by Alcpaone** [comment]
@PAO1908CH done? if so, please close the issue so we can all see the remaining tasks. Thank you!
